File: pyspider/examples/scrapy_style_demo.py
from pyspider.spider.spider import CrawlerProcess, FeedExporter, Item, Spider


class DemoSpider(Spider):
    name = "demo"
    start_urls = ["https://example.com"]

    def parse(self, page):
        yield Item(title=page.response.selector.title(), url=page.response.url)


if __name__ == "__main__":
    process = CrawlerProcess(DemoSpider())
    items = process.start()

    exporter = FeedExporter.json("artifacts/exports/pyspider-scrapy-demo.json")
    exporter.export_items(items)
    output = exporter.close()

    print(f"exported {len(items)} items to {output}")

File: javaspider/src/main/java/com/javaspider/examples/ScrapyStyleDemo.java
package com.javaspider.examples;

import com.javaspider.scrapy.CrawlerProcess;
import com.javaspider.scrapy.Spider;
import com.javaspider.scrapy.feed.FeedExporter;
import com.javaspider.scrapy.item.Item;

import java.util.List;

public final class ScrapyStyleDemo {
    private ScrapyStyleDemo() {
    }

    public static void main(String[] args) {
        Spider spider = new Spider() {
            {
                setName("demo");
                addStartUrl("https://example.com");
            }

            @Override
            public List<Object> parse(Response response) {
                return List.of(
                    new Item()
                        .set("title", response.selector().css("title").firstText())
                        .set("url", response.getUrl())
                );
            }
        };

        List<Item> items = new CrawlerProcess(spider).crawl();

        try (FeedExporter exporter = FeedExporter.json("artifacts/exports/javaspider-scrapy-demo.json")) {
            exporter.exportItems(items);
        }

        System.out.println("exported items: " + items.size());
    }
}
